packages/react/src/search-input/SearchInput.tsx

```diff
@@ -38,12 +49,13 @@ export const SearchInput = forwardRef<HTMLInputElement, SearchProps>(
 
                 forceValueChange(innerRef.current, "");
                 innerRef.current.focus();
+                onValueClear?.();
               }}
               {...styles.clear()}
             />
           )
         }
-        addonBefore={<IconMagnifyingGlass />}
+        addonBefore={withSearchIcon && <IconMagnifyingGlass />}
```

Comment:
i don't think we should provide a dedicated prop to clear the icon - consumers shouldn't be doing this 

and internally we can just send `addonBefore={null}` to override it ourselves